test/integration/suite/commaOnEnter.test.ts

```diff
@@ -0,0 +1,92 @@
+import * as vscode from 'vscode'
+
+import { expect } from 'chai'
+import { clearEditorText, stringWithPositions } from './utils'
+import dedent from 'string-dedent'
+
+describe('Comma on Enter', () => {
+    let document: vscode.TextDocument
+    let editor: vscode.TextEditor
+
+    // positions markers description:
+    // | - valid position to insert comma after Enter (adding empty newline below)
+    // $ - invalid position to insert comma after Enter (adding empty newline below)
+    // use editor text selection highlighting for faster navigation between positions
+    const FULL_FIXTURE = dedent/* json */ `
+    {
+        "key1": 43,/*$*/
+        "key2": 43/*|*/
+        /* key description */ "key3": "test"/*|*/ /* test comment */ // another commend
+        /*$*/
+        "key4": true/*|*/
+        "key4": false/*|*/
+        "key5": []/*|*/
+        "key6": {/*$*/
+        }/*|*/
+        "key7": "/*$*/ value",
```

Comment:
Didn't get idea of this check, line below will be inserted at this position (string remains unchanged). Imagine that `ctrl+Enter` is being done on `/*$*/` marker (not just `Enter`). I will fix it as following (as in this position comma could be potentially inserted):
```suggestion
        "key7": "value",/*$*/
```